hspec-discover/test/Test/Hspec/Discover/SortSpec.hs

```diff
@@ -0,0 +1,57 @@
+module Test.Hspec.Discover.SortSpec (main, spec) where
+
+import           Helper
+
+import           Data.List (sortBy)
+import           Data.Ord (comparing)
+
+import           Test.Hspec.Discover.Sort
+import           Test.QuickCheck
+
+main :: IO ()
+main = hspec spec
+
+
+spec :: Spec
+spec = do
+  describe "naturalSortKey" $ do
+    context "compares" $ do
+      it "reflexively" $ property $ \a -> do
+        a <=> a `shouldBe` EQ
+      it "symmetrically" $ property $ \a b -> do
+        a <=> b === compare EQ (b <=> a)
+      it "transitively" $ property $ \a b c -> do
+        a <=> b == b <=> c ==> a <=> b === a <=> c
+      it "empty first" $ property $ \a -> not (null a) ==> do
+        "" <=> a `shouldBe` LT
+        a <=> "" `shouldBe` GT
+      it "numbers first" $ do
+        "Hello2World" <=> "Hello World" `shouldBe` LT
+        "Hello World" <=> "Hello2World" `shouldBe` GT
+      it "string segments" $ do
+        "Hello2World9" <=> "Hello2World!0" `shouldBe` LT
+        "Hello2World!0" <=> "Hello2World9" `shouldBe` GT
+      it "numeric segments" $ do
+        "3.1.415" <=> "3.14.15" `shouldBe` LT
+        "3.14.15" <=> "3.1.415" `shouldBe` GT
+      it "numeric ties by string length" $ do
+        "Hello 02 World" <=> "Hello 002 World" `shouldBe` LT
+        "Hello 002 World" <=> "Hello 02 World" `shouldBe` GT
+      it "case squashing" $ do
+        "Helloworld1" <=> "HelloWorld2" `shouldBe` LT
+        "HelloWorld2" <=> "helloworld1" `shouldBe` GT
+      it "string ties by case" $ do
+        "HelloWorld3" <=> "Helloworld3" `shouldBe` LT
+        "Helloworld3" <=> "HelloWorld3" `shouldBe` GT
```

Comment:
https://github.com/hspec/hspec/commit/11837f69e345092897d3b360164f6b2f07747225

I think for derived `Ord` instances `a < b <=> b > a` is always true.